cabal-install/Distribution/Client/Dependency.hs

```diff
@@ -737,7 +745,16 @@ planPackagesProblems platform cinfo pkgs =
      | Configured pkg <- pkgs
      , let packageProblems = configuredPackageProblems platform cinfo pkg
      , not (null packageProblems) ]
-
+{-
+  --FIXME: this check is temporarily disabled because the use of nub below
+  -- to eliminate identical instances forces undefined fields used in the
+  -- solver testsuite. If the duplicates can be eliminated then this check
+  -- can be enable.
+  ++ [ DuplicatePackageSolverId (Graph.nodeKey (head dups)) dups
+     | let hackIdenticalInstances = filter ((>1) . length) . map nub
+     , dups <- hackIdenticalInstances --FIXME: should not have to do this
+             $ duplicatesBy (comparing Graph.nodeKey) pkgs ]
+-}
```

Comment:
@grayjay if this check is enabled we get lots of

> Exception: not yet configured: buildDepends

in the solver unit-tests. No doubt this is because of the `nub` above that forces the whole values, to check they're identical. But the bigger question is why we get duplicates in the first place.